src/Web/WsCommand/WsCmdGetDisp.cpp

```diff
@@ -77,38 +77,100 @@ void WsCmdGetDisp::execute(AsyncWebSocket* server, uint32_t clientId)
+            const uint8_t REPEAT_MAX = 0xFFU; /* Maximum repeat color counter value.       */
 
             DisplayMgr::getInstance().getFBCopy(framebuffer, fbLength, &slotId);
 
             preparePositiveResponse(msg);
-
             msg += slotId;
-            msg += DELIMITER;
-            msg += display.getWidth();
-            msg += DELIMITER;
-            msg += display.getHeight();
 
-            for(index = 0U; index <  fbLength; ++index)
+            /* RGB data is send in a "compressed" format using a repeat counter in
+             * the upper 8 bits. The send values are <rep>:<r>:<g>:<b>.
+             * The repeat counter indicates how often the same color shall be used
+             * in subsequent pixels. Use a small state machine to calculate the
+             * repeat counter.
+             *
+             * Example:
+             * A black only 32x8 framebuffer would be send as a single 0xFF000000 value.
+             *
+             */
+            enum
+            {
+                STATE_GETDISP_COLLECT = 0, /* Collect sequences of same color. */
+                STATE_GETDISP_SEND,        /* Send current color sequence.     */
+                STATE_GETDISP_FINISH       /* Terminate state machine          */
+            } state   = STATE_GETDISP_COLLECT;
+
+            lastColor = framebuffer[0];
+
+            while (state != STATE_GETDISP_FINISH)
             {
-                msg += DELIMITER;
-                msg += Util::uint32ToHex(framebuffer[index]);
+                if (STATE_GETDISP_COLLECT == state)
+                {
+                    if (index < fbLength)
+                    {
+                        color = framebuffer[index];
+                        if (color != lastColor)
+                        {
+                            /* Color has changed, send out current sequence */
+                            state = STATE_GETDISP_SEND;
+                        }
+                        else
+                        {
+                            ++repeat;
+                            if (REPEAT_MAX == repeat)
+                            {
+                                /* 8-bit repeat counter maximum reached, send color sequence. */
+                                state = STATE_GETDISP_SEND;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        /* End of frame buffer reached, send final color sequence. */
+                        state = STATE_GETDISP_SEND;
+                    }
+                    ++index;
+                }
+                else /* STATE_GETDISP_SEND */
+                {
+                    msg += DELIMITER;
+
+                    /* Lower 24 bits is RGB, upper 8 bits repeat count. */
+                    msg += Util::uint32ToHex(lastColor | (static_cast<uint32_t>(repeat) << 24));

```

Comment:
Decorator missing. Use uint32_t for repeat and kick the cast out.

Reply:
ok